src/ontology/config/prefixes.csv

```diff
@@ -227,6 +227,7 @@ ICD10CM,http://purl.bioontology.org/ontology/ICD10CM/
 ICD10CM2,https://icd.codes/icd10cm/
 ICD10WHO,https://icd.who.int/browse10/2019/en#/
 ICD10WHO2010,http://apps.who.int/classifications/icd10/browse/2010/en#/
+ICD11,http://purl.obolibrary.org/obo/mondo/mappings/unknown_prefix/ICD11/
```

Comment:
## ICD11 prefix
I used the one that we already had in `mondo.sssom.config.yml`, and added it to the following places:
- `config/prefixes.csv`
- `metadata/icd11.yml`

Reply:
Did you look at the resource? What kind of PURLs does it use?

Note that you will have to _very carefully_ seperate:

- ICD11 Foundation
- ICD11 linearisation

In this PR.

Reply:
Ahh, yes. I was doing some basic setup and looking at [the OWL](https://icd11files.blob.core.windows.net/tmp/whofic-2023-04-08.owl.gz) was gonna be my next step. But you're right, it's part of this.

Unfortunately, it looks like they don't declare any prefixes for their namespace(s):
<details><summary>Details</summary>
<p>

```owl
Prefix(:=<http://id.who.int/icd/schema/>)
Prefix(owl:=<http://www.w3.org/2002/07/owl#>)
Prefix(rdf:=<http://www.w3.org/1999/02/22-rdf-syntax-ns#>)
Prefix(xml:=<http://www.w3.org/XML/1998/namespace>)
Prefix(xsd:=<http://www.w3.org/2001/XMLSchema#>)
Prefix(rdfs:=<http://www.w3.org/2000/01/rdf-schema#>)
```

</p>
</details> 

I took a cursory look and I found at least 3 unique URI prefixes. Here they are below, with an indented example usage for each:
```
http://id.who.int/icd/entity/
  http://id.who.int/icd/entity/980862652
https://icd.who.int/dev11/f/en#/
  https://icd.who.int/dev11/f/en#/http%3A%2F%2Fid.who.int%2Ficd%2Fentity%2F1609069486
https://icdcdn.who.int/icd11referenceguide/en/html/index.html#
  https://icdcdn.who.int/icd11referenceguide/en/html/index.html#descriptions-related-to-transport-injury-events
```

@matentzn I'm worried about missing something. Can you think of a good way that I would be able to find a unique list of URI prefixes?
It seems tricky. Like I could parse the URL and end on the last `/` or `#` or maybe `_`... IDK.
Perhaps the best thing I can do is send an email? Whos' a good point person for this type of question?

Reply:
Interesting. Here is what I think:

```
ICD11_FOUNDATION: http://id.who.int/icd/entity/
```

`https://icd.who.int/dev11/f/en#/http%3A%2F%2Fid.who.int%2Ficd%2Fentity%2F1609069486` and `https://icdcdn.who.int/icd11referenceguide/en/html/index.html#` are not relevant as far as I can see right now, so probably no need for a prefix. I think they are just `linkouts` and do not refer to disease entities in the ontological sense.

> Can you think of a good way that I would be able to find a unique list of URI prefixes?

No its alway a bit manual. Sorry :( Print them in a list, sort them, eyeball them. That is what I usually do!

There is a method in curies you can try though that looks promising: https://curies.readthedocs.io/en/latest/discovery.html

> Perhaps the best thing I can do is send an email? Whos' a good point person for this type of question?
Or perhaps someone I could ask in the OBO slack?

I would avoid sending them emails for now for sociotechnological reasons, but the "sort" and "eyeball" method, perhaps combined with with `curies` discover method should do the job. If need be I can help you with this in the next call.

We are the first people working in detail with the foundation, so I dont think asking on slack will help either.

Reply:
Why don't you just convert the ontology to a supported format?

Reply:
RE ICD11 yes, I just meant to say to use icd11.foundation as prefix rather than icd11 because the latter is used for the linearisation.

Reply:
> Why don't you just convert the ontology to a supported format?

Yeah I don't know why I didn't just up and do that before, I'm gonna try that. I was already considering converting for another reason.

Reply:
- [ ] @matentzn evaluate and advise
<details><summary>URI prefix discovery results</summary>
<p>

OK. For some reason the discovery failed on RDF/XML but worked on TTL. Here's all that it found. Just 8 results.

### Non-ICD11 namespaces
http://www.w3.org/1999/02/22-rdf-syntax-ns#
http://www.w3.org/2000/01/rdf-schema#
http://www.w3.org/2002/07/owl#
http://www.w3.org/2004/02/skos/core#

### The one we chose so far for `ICD11_Foundation`
http://id.who.int/icd/entity/

### New ones
http://id.who.int/icd/schema/
http://who.int/
http://who.int/icd#Z_

#### http://who.int/icd#Z_
I don't know how best to describe this, but it is like a annotation property but not, and the only such case is `RetiredClasses`.
Also, I wonder if it's more appropriate to say that the URI prefix is `http://who.int/icd#` rather than `http://who.int/icd#Z_`.
```
cat cache/bak/icd11.owl/whofic-2023-04-08.owl | grep http://who.int/icd#Z_
Declaration(Class(<http://who.int/icd#Z_RetiredClasses>))
SubClassOf(<http://id.who.int/icd/entity/1220239748> <http://who.int/icd#Z_RetiredClasses>)
SubClassOf(<http://id.who.int/icd/entity/1975819343> <http://who.int/icd#Z_RetiredClasses>)
SubClassOf(<http://id.who.int/icd/entity/1998311477> <http://who.int/icd#Z_RetiredClasses>)
SubClassOf(<http://id.who.int/icd/entity/434164741> <http://who.int/icd#Z_RetiredClasses>)
SubClassOf(<http://id.who.int/icd/entity/659906204> <http://who.int/icd#Z_RetiredClasses>)
```

#### http://who.int/
Looks like this is not actually a URI prefix. It's just part of the above case `http://who.int/icd#Z_`
```
cat cache/bak/icd11.owl/whofic-2023-04-08.owl | grep http://who.int/
Ontology(<http://who.int/icd>
Declaration(Class(<http://who.int/icd#Z_RetiredClasses>))
SubClassOf(<http://id.who.int/icd/entity/1220239748> <http://who.int/icd#Z_RetiredClasses>)
SubClassOf(<http://id.who.int/icd/entity/1975819343> <http://who.int/icd#Z_RetiredClasses>)
SubClassOf(<http://id.who.int/icd/entity/1998311477> <http://who.int/icd#Z_RetiredClasses>)
SubClassOf(<http://id.who.int/icd/entity/434164741> <http://who.int/icd#Z_RetiredClasses>)
SubClassOf(<http://id.who.int/icd/entity/659906204> <http://who.int/icd#Z_RetiredClasses>)
```

#### http://id.who.int/icd/schema/
I don't know if we'll have a use for this one, but we could call it `ICD11_SCHEMA`. IDK if you need to see the detail, but:
```
cat cache/bak/icd11.owl/whofic-2023-04-08.owl | grep http://id.who.int/icd/schema/
Prefix(:=<http://id.who.int/icd/schema/>)
# Object Property: <http://id.who.int/icd/schema/activitiesDescriptor> (<http://id.who.int/icd/schema/activitiesDescriptor>)
# Object Property: <http://id.who.int/icd/schema/activityWhenInjured> (<http://id.who.int/icd/schema/activityWhenInjured>)
# Object Property: <http://id.who.int/icd/schema/anatomy> (<http://id.who.int/icd/schema/anatomy>)
# Object Property: <http://id.who.int/icd/schema/biologicalIndicator> (<http://id.who.int/icd/schema/biologicalIndicator>)
# Object Property: <http://id.who.int/icd/schema/causality> (<http://id.who.int/icd/schema/causality>)
# Object Property: <http://id.who.int/icd/schema/chemicalAgent> (<http://id.who.int/icd/schema/chemicalAgent>)
# Object Property: <http://id.who.int/icd/schema/consciousnessMeasure> (<http://id.who.int/icd/schema/consciousnessMeasure>)
# Object Property: <http://id.who.int/icd/schema/counterMeasureDescriptor> (<http://id.who.int/icd/schema/counterMeasureDescriptor>)
# Object Property: <http://id.who.int/icd/schema/counterPartInCollision> (<http://id.who.int/icd/schema/counterPartInCollision>)
# Object Property: <http://id.who.int/icd/schema/course> (<http://id.who.int/icd/schema/course>)
# Object Property: <http://id.who.int/icd/schema/customBurnScaleValue> (<http://id.who.int/icd/schema/customBurnScaleValue>)
# Object Property: <http://id.who.int/icd/schema/diagnosisConfirmedBy> (<http://id.who.int/icd/schema/diagnosisConfirmedBy>)
# Object Property: <http://id.who.int/icd/schema/distribution> (<http://id.who.int/icd/schema/distribution>)
# Object Property: <http://id.who.int/icd/schema/durationOfComa> (<http://id.who.int/icd/schema/durationOfComa>)
# Object Property: <http://id.who.int/icd/schema/etiology> (<http://id.who.int/icd/schema/etiology>)
# Object Property: <http://id.who.int/icd/schema/externalCauseDimension> (<http://id.who.int/icd/schema/externalCauseDimension>)
# Object Property: <http://id.who.int/icd/schema/fractureSubtype> (<http://id.who.int/icd/schema/fractureSubtype>)
# Object Property: <http://id.who.int/icd/schema/genomicAndChomosomalAnomaly> (<http://id.who.int/icd/schema/genomicAndChomosomalAnomaly>)
# Object Property: <http://id.who.int/icd/schema/hasGCSEyeScore> (<http://id.who.int/icd/schema/hasGCSEyeScore>)
# Object Property: <http://id.who.int/icd/schema/hasGCSMotorScore> (<http://id.who.int/icd/schema/hasGCSMotorScore>)
# Object Property: <http://id.who.int/icd/schema/hasGCSScore> (<http://id.who.int/icd/schema/hasGCSScore>)
# Object Property: <http://id.who.int/icd/schema/hasGCSVerbalScore> (<http://id.who.int/icd/schema/hasGCSVerbalScore>)
# Object Property: <http://id.who.int/icd/schema/hasPupilReactionScore> (<http://id.who.int/icd/schema/hasPupilReactionScore>)
# Object Property: <http://id.who.int/icd/schema/hasSeverity> (<http://id.who.int/icd/schema/hasSeverity>)
# Object Property: <http://id.who.int/icd/schema/histopathology> (<http://id.who.int/icd/schema/histopathology>)
# Object Property: <http://id.who.int/icd/schema/infectiousAgent> (<http://id.who.int/icd/schema/infectiousAgent>)
# Object Property: <http://id.who.int/icd/schema/injuryQualifier> (<http://id.who.int/icd/schema/injuryQualifier>)
# Object Property: <http://id.who.int/icd/schema/intent> (<http://id.who.int/icd/schema/intent>)
# Object Property: <http://id.who.int/icd/schema/intentDescriptor> (<http://id.who.int/icd/schema/intentDescriptor>)
# Object Property: <http://id.who.int/icd/schema/laterality> (<http://id.who.int/icd/schema/laterality>)
# Object Property: <http://id.who.int/icd/schema/levelOfConsciousness> (<http://id.who.int/icd/schema/levelOfConsciousness>)
# Object Property: <http://id.who.int/icd/schema/mechanismOfInjury> (<http://id.who.int/icd/schema/mechanismOfInjury>)
# Object Property: <http://id.who.int/icd/schema/medication> (<http://id.who.int/icd/schema/medication>)
# Object Property: <http://id.who.int/icd/schema/modeOfTransport> (<http://id.who.int/icd/schema/modeOfTransport>)
# Object Property: <http://id.who.int/icd/schema/objectOrSubstanceProducingInjury> (<http://id.who.int/icd/schema/objectOrSubstanceProducingInjury>)
# Object Property: <http://id.who.int/icd/schema/placeOfOccurrence> (<http://id.who.int/icd/schema/placeOfOccurrence>)
# Object Property: <http://id.who.int/icd/schema/placeOfOccurrenceDescriptor> (<http://id.who.int/icd/schema/placeOfOccurrenceDescriptor>)
# Object Property: <http://id.who.int/icd/schema/postcoordinationAxis> (<http://id.who.int/icd/schema/postcoordinationAxis>)
# Object Property: <http://id.who.int/icd/schema/regional> (<http://id.who.int/icd/schema/regional>)
# Object Property: <http://id.who.int/icd/schema/relational> (<http://id.who.int/icd/schema/relational>)
# Object Property: <http://id.who.int/icd/schema/serotype> (<http://id.who.int/icd/schema/serotype>)
# Object Property: <http://id.who.int/icd/schema/settingOfEvent> (<http://id.who.int/icd/schema/settingOfEvent>)
# Object Property: <http://id.who.int/icd/schema/specificAnatomy> (<http://id.who.int/icd/schema/specificAnatomy>)
# Object Property: <http://id.who.int/icd/schema/substanceUse> (<http://id.who.int/icd/schema/substanceUse>)
# Object Property: <http://id.who.int/icd/schema/temporalPatternAndOnset> (<http://id.who.int/icd/schema/temporalPatternAndOnset>)
# Object Property: <http://id.who.int/icd/schema/temporality> (<http://id.who.int/icd/schema/temporality>)
# Object Property: <http://id.who.int/icd/schema/timeInLife> (<http://id.who.int/icd/schema/timeInLife>)
# Object Property: <http://id.who.int/icd/schema/topology> (<http://id.who.int/icd/schema/topology>)
# Object Property: <http://id.who.int/icd/schema/transportDimension> (<http://id.who.int/icd/schema/transportDimension>)
# Object Property: <http://id.who.int/icd/schema/transportUserRoleOfInjuredPerson> (<http://id.who.int/icd/schema/transportUserRoleOfInjuredPerson>)
# Object Property: <http://id.who.int/icd/schema/typeOfInjury> (<http://id.who.int/icd/schema/typeOfInjury>)
```



http://id.who.int/icd/entity/



</p>
</details>

Reply:
- [ ] @matentzn Consider and advise on prefix case
You mentioned using `ICD11_FOUNDATION`

I prefer `ICD11_Foundation` because I'm not aware of any conventions, and this leaves us open to more flexibility in the future, should we find different things we want to sub-namespace, because we have access to both appending with `_` or also using camelCase.

Reply:
Ok cool, thanks for running `curies.discover()`!

Rather than debating now the exact prefixes, which are only relevant for now for us internally, lets use for now

```
rdf: http://www.w3.org/1999/02/22-rdf-syntax-ns#
rdfs: http://www.w3.org/2000/01/rdf-schema#
owl: http://www.w3.org/2002/07/owl#
skos: http://www.w3.org/2004/02/skos/core#
icd11.foundation: http://id.who.int/icd/entity/
icd11.schema: http://id.who.int/icd/schema/
icd11.z: http://who.int/icd#Z_
```

which is the conventions that bioregistry uses. We can change the prefixes later if there is a reason (but since we are not sharing anything, it does not really matter).

For sure will we want both the schema and the "Z" thing (we can determine later why it is so weird), because they can be mapped to RO properties in the future. 

It would be good to get a sense of which entities are not mapped after these 7 namespaces are applied.

Reply:
Ok, sounds good.
The only remaining task in here is:
> It would be good to get a sense of which entities are not mapped after these 7 namespaces are applied.

So I'm going to resolve this thread and [open a new one](https://github.com/monarch-initiative/mondo-ingest/pull/412#discussion_r1485654294) with that objective.